operator/controller/src/main/java/io/apicurio/registry/operator/resource/app/AppDeploymentResource.java

```diff
@@ -36,22 +38,36 @@ protected Deployment desired(ApicurioRegistry3 primary, Context<ApicurioRegistry
 
         var d = APP_DEPLOYMENT_KEY.getFactory().apply(primary);
 
-        var appEnv = new ArrayList<>(List.of(
-                // spotless:off
-                new EnvVarBuilder().withName("QUARKUS_PROFILE").withValue("prod").build(),
-                new EnvVarBuilder().withName("APICURIO_CONFIG_CACHE_ENABLED").withValue("true").build(),
-                new EnvVarBuilder().withName("QUARKUS_HTTP_ACCESS_LOG_ENABLED").withValue("true").build(),
-                new EnvVarBuilder().withName("QUARKUS_HTTP_CORS_ORIGINS").withValue("*").build(),
-                new EnvVarBuilder().withName("APICURIO_REST_DELETION_GROUP_ENABLED").withValue("true").build(),
-                new EnvVarBuilder().withName("APICURIO_REST_DELETION_ARTIFACT_ENABLED").withValue("true").build(),
-                new EnvVarBuilder().withName("APICURIO_REST_DELETION_ARTIFACTVERSION_ENABLED").withValue("true").build(),
-                new EnvVarBuilder().withName("APICURIO_APIS_V2_DATE_FORMAT").withValue("yyyy-MM-dd''T''HH:mm:ssZ").build()
-                // spotless:on
-        ));
-
-        d.getSpec().getTemplate().getSpec().getContainers().get(0).setEnv(appEnv);
+        var envVars = new LinkedHashMap<String, EnvVar>();
+
+        // spotless:off
+        addEnvVar(envVars, new EnvVarBuilder().withName("QUARKUS_PROFILE").withValue("prod").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("APICURIO_CONFIG_CACHE_ENABLED").withValue("true").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("QUARKUS_HTTP_ACCESS_LOG_ENABLED").withValue("true").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("QUARKUS_HTTP_CORS_ORIGINS").withValue("*").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("APICURIO_REST_DELETION_GROUP_ENABLED").withValue("true").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("APICURIO_REST_DELETION_ARTIFACT_ENABLED").withValue("true").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("APICURIO_REST_DELETION_ARTIFACTVERSION_ENABLED").withValue("true").build());
+        addEnvVar(envVars, new EnvVarBuilder().withName("APICURIO_APIS_V2_DATE_FORMAT").withValue("yyyy-MM-dd''T''HH:mm:ssZ").build());
+        // spotless:on
+
+        // This must be done after any modification of the map by the operator.
+        primary.getSpec().getApp().getEnv().forEach(e -> {
```

Comment:
I find it a bit hard to follow the logic, what do you think about building directly the final map?

Something like:
```java
var envVars = new LinkedHashMap<String, EnvVar>();
for (var e: primary.getSpec().getApp().getEnv()) {
    envVars.put(e.getName(), e);
}
```
and change `addEnvVar` by adding:
```java
if (!map.containsKey(envVar.getName())) {
```

am I missing any invariant?